Packages/MIES/MIES_Utilities_File.ipf

```diff
@@ -222,96 +222,175 @@ Function/S GetUniqueSymbolicPath([string prefix])
+
+	if(ParamIsDefault(extension))
+		WAVE/ZZ/T files = GetAllFilesRecursivelyFromPath(pathName)
+	else
+		WAVE/ZZ/T files = GetAllFilesRecursivelyFromPath(pathName, regex = "\"" + extension + "$")
+	endif
+
+	return TextWaveToList(files, FILE_LIST_SEP)
+End
+
 /// @brief Return a list of all files from the given symbolic path
-///        and its subfolders. The list is pipe (`FILE_LIST_SEP`) separated as
-///        the semicolon (`;`) is a valid character in filenames.
-///
-/// Note: This function does *not* work on MacOSX as there filenames are allowed
-///       to have pipe symbols in them.
+///        and its subfolders.
 ///
 /// @param pathName igor symbolic path to search recursively
-/// @param extension [optional, defaults to all files] file suffixes to search for
-Function/S GetAllFilesRecursivelyFromPath(string pathName, [string extension])
+/// @param regex [optional, defaults to all files] regular expression to match the absolute file path against
+/// @param resolveAliases [optional, defaults to false] attempt to resolve aliases/shortcuts (slows down execution!)
+Function/WAVE GetAllFilesRecursivelyFromPath(string pathName, [string regex, variable resolveAliases])
 
-	string fileOrPath, folders, subFolderPathName, fileName
-	string files, allFilesList
-	string allFiles         = ""
-	string foldersFromAlias = ""
-	variable err, isWindows
+	string files, subFoldersList, subFilesList, cdf, workPath
+	variable err
+	variable numFolders, numFiles, needsAliasResolving, i, needsFiltering
 
-#ifdef WINDOWS
-	isWindows = 1
-#endif // WINDOWS
+	if(ParamIsDefault(regex))
+		regex = ".*"
+	else
+		ASSERT(IsValidRegexp(regex), "Expected a valid regex")
+		needsFiltering = 1
+	endif
+
+	if(ParamIsDefault(resolveAliases))
+		resolveAliases = 0
+	else
+		resolveAliases = !!resolveAliases
+	endif
+
+	Make/FREE/N=(MINIMUM_WAVE_SIZE_LARGE)/T resultFiles
+	SetNumberInWaveNote(resultFiles, NOTE_INDEX, 0)
+
+	Make/FREE/N=(MINIMUM_WAVE_SIZE)/T folders
+	SetNumberInWaveNote(folders, NOTE_INDEX, 0)
 
 	PathInfo $pathName
 	ASSERT(V_flag, "Given symbolic path does not exist")
+	Make/T/FREE startFolder = {S_path}
+	numFolders = ConcatenateWavesWithNoteIndex(folders, startFolder)
 
-	if(ParamIsDefault(extension))
-		extension = "????"
-	endif
+	workPath = GetUniqueSymbolicPath()
 
 	AssertOnAndClearRTError()
-	allFilesList = IndexedFile($pathName, -1, extension, "????", FILE_LIST_SEP); err = GetRTError(1)
+	for(i = 0; i < numFolders; i += 1)
 
-	if(isWindows && cmpstr(extension, "????") && cmpstr(extension, ".lnk"))
-		allFiles = AddPrefixToEachListItem(S_path, allFilesList, sep = FILE_LIST_SEP)
-	else
-		// try to resolve aliases/shortcuts when we can have them
-		WAVE/T allFilesInDir = ListToTextWave(allFilesList, FILE_LIST_SEP)
-		for(fileName : allFilesInDir)
+		cdf = folders[i]
+		NewPath/Q/O/Z $workPath, cdf
 
-			GetFileFolderInfo/P=$pathName/Q/Z fileName
+		subFoldersList = IndexedDir($workPath, -1, 1, FILE_LIST_SEP); err = GetRTError(1)
```

Comment:
Add a
`subFoldersList = ""`
before this line that in case of a caught RTE the string has not the old content.

Also for the IndexedFile call.